### PR TITLE
MAINT: Reorganize modules and cache

### LINKED ISSUE #616: MAINT: Proposed minor reorg
Looking through the scripts, I notice a couple of things I'd like to reorganize. Proposal:

1. Rename `source/_03_make_cov.py` -> `sensor/_05_make_cov.py`. Covariances are useful for more than just source space -- I've used them for `plot_white` and decoding (`XDawn`) as well before doing any source analyses. They are not dependent in any way on any source space "stuff" (bem/trans/fwd/inv).
2. Move BEM solution computation from `source/_02_make_forward.py` to `source/_01_make_bem.py` (renamed from `source/_01_make_bem_surfaces.py`. In general the BEM surface <-> BEM solution relationship should be 1:1 (the second completely dependent on the first, and should be consistent with it). The forward step depends on the solution only (not the surfaces alone) so it makes sense to separate it.

If this sounds reasonable I can implement it in https://github.com/mne-tools/mne-bids-pipeline/pull/615

### PR DESCRIPTION
**Pull request by larsoner**
### Before merging …

- [x] Changelog has been updated (`docs/source/changes.md`)
- [x] Reorg according to #616
- [x] Add source space caching
- [x] Add forward caching
- [x] Add inverse caching

I ended up adding caching of the forward and inverse steps while reorganizing since I needed to do a bit of filename mangling anyway. Those changes are pretty easily visible separate from the red lines in `make_forward.py` due to refactoring.

Closes #616

### CODE REVIEW

**Comment by larsoner** [comment]
... and it turns out that `ds003107` has the BEM surfaces already but not the BEM solution. So I split that into two steps, which seems fine

File: config.py [hunk 1]
```diff
@@ -1966,7 +1968,12 @@ def noise_cov(bids_path):
     },
     'run': {
         'bold': True
-    }
+    },
+    'box': {
+        'color': 'cyan',
+        'bold': True,
+        'bright': True,
```
[the remark this answers is not in the document]
Comment (larsoner) [reply config.py:1975]: I just tried to follow the existing ways of doing things. I'm not too familiar with coloredlogs. Maybe in some follow-up PR that changes/tweaks it everywhere?

[the remark this answers is not in the document]
Comment (hoechenberger) [reply config.py:1975]: Ok!

**Review by agramfort — APPROVED** [review]

**Comment by agramfort** [comment]
thx @larsoner

**Pull request merged**